Source/santad/SNTPolicyProcessor.h

```diff
@@ -51,6 +51,13 @@
                           (NSDictionary *_Nullable (^_Nonnull)(
                             const char *_Nullable teamID,
                             NSDictionary *_Nullable entitlements))entitlementsFilterCallback;
+- (nonnull SNTCachedDecision *)decisionForFileInfo:(nonnull SNTFileInfo *)fileInfo
+                                     targetProcess:(nonnull const es_process_t *)targetProc
+                          preCodesignCheckCallback:(void (^_Nullable)(void))preCodesignCheckCallback
```

Comment:
Adding an additional callback here is starting to get unwieldy as an interface. Consider refactoring these callbacks out into a delegate protocol. Then have a new concrete object that implements the protocol to handle the callbacks.

Reply:
This would be a pretty large refactor. Reckon i'm overall not a huge fan of the delegate pattern, but it's a fair point about multiple callbacks becoming unwieldy. Part of the problem is there are multiple paths into the PolicyProcessor. One from the main AUTH EXEC processing flow, another via XPC/santactl. Obviously delegate could handle this as well by jsut not setting the delegate property via the XPC route. But that feels odd too. Having the interfaces into the PolicyProcessor makes it more clear (to me) about a caller's intention, without having to search around for when/how random delegates get set.

I'm inclined to punt on the major refactor to delegates, but I'm open to other ideas of how to simplify this.

Reply:
At its core the ObjC delegate pattern isn't terrible in my opinion. It is ubiquitous and pretty well understood. But if we were to refactor maybe we punt and refactor in C++. There are similar "delegate/observer" patterns we could use.

https://chromium.googlesource.com/chromium/src/+/HEAD/docs/patterns/inversion-of-control.md